homeassistant/components/sum/sensor.py

```diff
@@ -0,0 +1,214 @@
+"""Support for displaying sum values."""
+from __future__ import annotations
+
+from collections.abc import Mapping
+import logging
+from typing import Any
+
+from homeassistant.components.sensor import SensorEntity, SensorStateClass
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    ATTR_UNIT_OF_MEASUREMENT,
+    CONF_NAME,
+    CONF_UNIQUE_ID,
+    STATE_UNAVAILABLE,
+    STATE_UNKNOWN,
+)
+from homeassistant.core import Event, HomeAssistant, State, callback
+from homeassistant.helpers import entity_registry as er
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.event import async_track_state_change_event
+from homeassistant.helpers.reload import async_setup_reload_service
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType, EventType
+
+from .const import CONF_ENTITY_IDS, CONF_ROUND_DIGITS, DOMAIN, PLATFORMS
+
+_LOGGER = logging.getLogger(__name__)
+
+ICON = "mdi:calculator"
+ATTR_ENTITIES = "entities"
+ATTR_VALID_ENTITIES = "valid_entities"
+ATTR_COUNT_VALID = "count_valid"
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Initialize sum config entry."""
+    registry = er.async_get(hass)
+    entity_ids = er.async_validate_entity_ids(
+        registry, config_entry.options[CONF_ENTITY_IDS]
+    )
+    round_digits = int(config_entry.options[CONF_ROUND_DIGITS])
+
+    async_add_entities(
+        [
+            SumSensor(
+                entity_ids,
+                config_entry.title,
+                round_digits,
+                config_entry.entry_id,
+            )
+        ]
+    )
+
+
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    async_add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Set up the Sum sensor."""
+    if discovery_info is None:
+        return
+    entity_ids: list[str] = discovery_info[CONF_ENTITY_IDS]
+    name: str = discovery_info[CONF_NAME]
+    round_digits: float = discovery_info[CONF_ROUND_DIGITS]
+    unique_id: str | None = discovery_info.get(CONF_UNIQUE_ID)
+
+    await async_setup_reload_service(hass, DOMAIN, PLATFORMS)
+
+    async_add_entities([SumSensor(entity_ids, name, round_digits, unique_id)])
+
+
+def calc_sum(sensor_values: list[tuple[str, Any]], round_digits: int) -> float | None:
+    """Calculate sum value, skipping unknown/unavailable state."""
+    result = []
+    for _, sensor_value in sensor_values:
+        if sensor_value in [STATE_UNKNOWN, STATE_UNAVAILABLE]:
+            continue
```

Comment:
I'm not sure if this is the right thing to do. If I want to sum two sensors, and one isn't available, I think it would make sense if this sense wasn't available either.

Otherwise, the values can become quite unexpected.